src/main/java/gyro/aws/kms/KmsKeyResource.java

```diff
@@ -524,4 +583,30 @@ private String getProcessedPolicy(String policy) {
             throw new GyroException(String.format("Could not read the json `%s`",policy),ex);
         }
     }
+
+    @Override
+    public List<ValidationError> validate(Set<String> configuredFields) {
+        List<ValidationError> errors = new ArrayList<>();
+
+        if (configuredFields.contains("primary-kms-key")) {
+            if (!configuredFields.contains("primary-key-region")) {
+                errors.add(new ValidationError(this, null, "primary-key-region is required when primary-kms-key is set."));
+            }
+            if (configuredFields.contains("multi-region")) {
+                errors.add(new ValidationError(this, null, "replica-key cannot be multi-region."));
+            }
+            if (configuredFields.contains("key-rotation")) {
+                errors.add(new ValidationError(this, null, "replica-key cannot enable key rotation."));
+            }
+            if (configuredFields.contains("pending-window")) {
+                errors.add(new ValidationError(this, null, "replica-key cannot have pending window."));
+            }
+        } else {
+            if (configuredFields.contains("primary-kms-key")) {
+                errors.add(new ValidationError(this, null, "primary-kms-key is required when primary-key-region is set."));
+            }
+        }
```

Comment:
I think you meant `primary-key-region` here
```suggestion
        } else if (configuredFields.contains("primary-key-region")) {
            errors.add(new ValidationError(this, null, "primary-kms-key is required when primary-key-region is set."));
        }
```